packages/amplify-category-geo/src/service-utils/mapUtils.ts

```diff
@@ -55,6 +55,7 @@ export const modifyMapResource = async (context: $TSContext, parameters: MapPara
   const paramsToUpdate = ['accessType', 'pricingPlan'];
   paramsToUpdate.forEach(param => {
     context.amplify.updateamplifyMetaAfterResourceUpdate(category, parameters.name, param, (mapMetaParameters as $TSObject)[param]);
+    context.amplify.updateBackendConfigAfterResourceUpdate(category, parameters.name, param, (mapMetaParameters as $TSObject)[param]);
```

Comment:
@phani-srikar This is where the `checkout` generating wrong meta will be fixed. In CLI, when you checkout a env without a `--restore` flag, the new meta is generated based on the `backend-config.json`. The code snippet is as following.
https://github.com/aws-amplify/amplify-cli/blob/60255ee718f8d047ee665237db61d357a24fc41c/packages/amplify-cli/src/initialize-env.ts#L28-L31

https://github.com/aws-amplify/amplify-cli/blob/60255ee718f8d047ee665237db61d357a24fc41c/packages/amplify-cli/src/initialize-env.ts#L112-L116

However, the `backend-config.json` is not updated unless you run `add` command. That's why there are two default maps when running checkout command. Therefore, adding the updating backend config process after geo update will solve the problem, which will generate the right backend config and amplify meta as well